miniapps/tools/gridfunction-bounds.cpp

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) 2010-2025, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+//
+//    ---------------------------------------------------------------------
+//            Compute bounds of the given grid-function
+//    ---------------------------------------------------------------------
+//
+// This miniapp computes piecewise linear bounds on a given gridfunction, and
+// visualizes the lower and upper bound for each element. The bounding approach
+// is based on the method described in:
+//
+// (1) Section 3 of Mittal et al., "General Field Evaluation in High-Order
+//     Meshes on GPUs"
+// and
+// (2) Dzanic et al., "A method for bounding high-order finite element
+//     functions: Applications to mesh validity and bounds-preserving limiters".
+//
+//
+// Compile with: make gridfunction-bounds
+//
+// Sample runs:
+// mpirun -np 4 gridfunction-bounds
+// mpirun -np 4 gridfunction-bounds -nb 100 -ref 5 -bt 2 -l2
+
+#include "mfem.hpp"
+#include <iostream>
+#include <fstream>
+
+using namespace mfem;
+using namespace std;
+
+void VisualizeField(ParMesh &pmesh, ParGridFunction &input,
+                    char *title, int pos_x, int pos_y);
+
+int main (int argc, char *argv[])
+{
+   // 0. Initialize MPI and HYPRE.
+   Mpi::Init(argc, argv);
+   int myid = Mpi::WorldRank();
```

Comment:
As in previous comment, can replace the `myid == 0` checks with `Mpi::Root()` and then remove this variable